### ISSUE TITLE
After clusterrole/role is created, yaml shows resource item is uppercase

### ISSUE THREAD

**Issue by chrisho** [rancher/rancher#38068]
**Rancher Server Setup**
- Rancher version: `v2.6.5`
- Installation option (Docker install/Helm Chart): docker install
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): k3s
     Node Setup: 1 node
     Version: `v1.23.6+k3s1`
- Proxy/Cert Details:  `self-signed`

**Information about the Cluster**
- Kubernetes version: `v1.23.7+rke2r2`
- Cluster Type (Local/Downstream): Local

**User Information**
- What is the role of the user logged in? Admin
~  - If custom, define the set of permissions:~

**Describe the bug**
After clusterrole/role is created, yaml shows that resource item is uppercase, it should be lowercase. 
eg:  `Node` is selected, yaml shows `Node` too

**To Reproduce**
1. create clusterrole/role page.
2. select any verbs and any resources, then click `save`.
3. click `edit yaml`.

**Result**
```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  creationTimestamp: "2022-06-23T02:31:22Z"
  managedFields:
  - apiVersion: rbac.authorization.k8s.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:rules: {}
    manager: rancher
    operation: Update
    time: "2022-06-23T02:31:22Z"
  name: test
  resourceVersion: "2423173"
  uid: c53fa817-f06c-4670-bc07-395188ccd2fb
rules:
- apiGroups:
  - ""
  resources:
  - Node
  verbs:
  - list
  - update

```

**Expected Result**
resources item must comply with k8s rbac rules, eg: after Node is selected, yaml shows and requests to rancher is `nodes`.

**Screenshots**
<img width="863" alt="image" src="https://user-images.githubusercontent.com/3347460/175198567-fae05603-9ff1-43aa-8e67-b80f9e003b8e.png">
<img width="729" alt="image" src="https://user-images.githubusercontent.com/3347460/175198607-0a87e8fc-c9cf-4c01-8bbd-09af9a2243ff.png">

**Additional context**

**Comment by richard-cox** [comment]
Possible duplicate of https://github.com/rancher/dashboard/issues/6090

**Comment by catherineluse** [comment]
Yes, it is a duplicate. Cody fixed it in his PR https://github.com/rancher/dashboard/pull/6163. The role creation forms add all resources in lowercase now.
